apps/etterlatte-vedtaksvurdering-kafka/src/main/kotlin/MigreringHendelser.kt

```diff
@@ -0,0 +1,83 @@
+import no.nav.etterlatte.rapidsandrivers.EventNames
+import no.nav.etterlatte.rapidsandrivers.EventNames.FATT_VEDTAK
+import no.nav.etterlatte.rapidsandrivers.EventNames.OPPRETT_VEDTAK
+import no.nav.etterlatte.rapidsandrivers.migrering.Migreringshendelser.VEDTAK
+import no.nav.etterlatte.regulering.VedtakService
+import no.nav.helse.rapids_rivers.JsonMessage
+import no.nav.helse.rapids_rivers.MessageContext
+import no.nav.helse.rapids_rivers.RapidsConnection
+import no.nav.helse.rapids_rivers.River
+import org.slf4j.LoggerFactory
+import rapidsandrivers.BEHANDLING_ID_KEY
+import rapidsandrivers.SAK_ID_KEY
+import rapidsandrivers.behandlingId
+import rapidsandrivers.sakId
+import rapidsandrivers.withFeilhaandtering
+import java.util.*
+
+internal class MigreringHendelser(
+    rapidsConnection: RapidsConnection,
+    private val vedtak: VedtakService
+) : River.PacketListener {
+    private val logger = LoggerFactory.getLogger(this::class.java)
+
+    init {
+        River(rapidsConnection).apply {
+            eventName(VEDTAK)
+            validate { it.requireKey(BEHANDLING_ID_KEY) }
+            validate { it.requireKey(SAK_ID_KEY) }
+            correlationId()
+        }.register(this)
+    }
+
+    override fun onPacket(packet: JsonMessage, context: MessageContext) =
+        withLogContext(packet.correlationId) {
+            val behandlingId = packet.behandlingId
+            val sakId = packet.sakId
+            logger.info("Oppretter vedtak for migrer behandling $behandlingId")
+
+            withFeilhaandtering(packet, context, OPPRETT_VEDTAK) {
+                val respons = vedtak.upsertVedtak(behandlingId)
+                logger.info("Opprettet vedtak ${respons.vedtakId} for migrert behandling: $behandlingId")
+                packet[SKAL_SENDE_BREV] = false
+            }
+                .takeIf { it.isSuccess }
+                ?.let {
+                    fattVedtak(packet, context, behandlingId, sakId)
+                        .takeIf { it.isSuccess }
+                        ?.let { attester(packet, context, behandlingId, sakId) }
+                }
+        }
+
+    private fun fattVedtak(
+        packet: JsonMessage,
+        context: MessageContext,
+        behandlingId: UUID,
+        sakId: Long
+    ) = withFeilhaandtering(packet, context, FATT_VEDTAK) {
+        val fattetVedtak = vedtak.fattVedtak(behandlingId)
+        logger.info(
+            "Fattet vedtak ${fattetVedtak.vedtakId} for sak: $sakId og behandling: $behandlingId"
+        )
+    }
+
+    private fun attester(
+        packet: JsonMessage,
+        context: MessageContext,
+        behandlingId: UUID,
+        sakId: Long
+    ) {
+        withFeilhaandtering(packet, context, EventNames.ATTESTER) {
+            val attestert = vedtak.attesterVedtak(behandlingId)
+            logger.info(
+                "Attesterte vedtak ${attestert.vedtakId} for sak: $sakId og behandling: $behandlingId"
+            )
+        }
+    }
```

Comment:
Jeg lurer på om vi kanskje bør ha en egen flyt for automatiske behandlinger, så den ikke fatter og attesterer, men heller er igjennom noe litt eget (automatisk fattet og attestert f.eks.?) -- siden den i prinsippet ikke er igjennom en totrinnskontroll virker det litt rart å simulere det slik. Men ikke noe som er nødvendig nå, mest noe jeg tenkte på når jeg så flyten.

Reply:
Ja, har også vore innom den tanken. Det samme prinsippet gjeld jo også for regulering.